src/redux/session.js

```diff
@@ -103,3 +104,46 @@ module.exports.refreshSession = () => (dispatch => {
         return;
     });
 });
+
+module.exports.refreshSessionWithRetry = () => (dispatch => {
```

Comment:
I left `refreshSession()` intact, as opposed to refactoring the shared code out, because I wanted this PR to not be too complicated. If we go in this direction, we could refactor in a follow-up PR, perhaps by moving both calls to `api()` into a single function in lib/session.js, and perhaps by moving the various redirects into a separate function in lib/session.js or somewhere else.